File: app/src/main/java/ru/nikitaboiko/instagramgallery/di/AppModule.java
package ru.nikitaboiko.instagramgallery.di;

import dagger.Module;
import dagger.Provides;
import ru.nikitaboiko.instagramgallery.App;

@Module
public class AppModule {
    private App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Provides
    public App app() {
        return app;
    }
}

File: app/src/main/java/com/example/instagramgallery/di/AppModule.kt
package com.example.instagramgallery.di

import com.example.instagramgallery.App
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Singleton
@Module
class AppModule(private val app: App) {

    @Provides
    fun app(): App {
        return app
    }
}